client/driver/exec.go

```diff
@@ -41,10 +47,40 @@ func (d *ExecDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool,
 }
 
 func (d *ExecDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle, error) {
-	// Get the command
+	// Get the command to be ran, or if omitted, download an artifact for
+	// execution. Currently a supplied command takes precedence, and an artifact
+	// is only downloaded if no command is supplied
 	command, ok := task.Config["command"]
 	if !ok || command == "" {
-		return nil, fmt.Errorf("missing command for exec driver")
+		source, sok := task.Config["artifact_source"]
```

Comment:
This doesn't seem like the right behavior. What if my artifact is a tar.gz containing all my assets.Then I would want to download it and then untar and run it.